darts/tests/models/forecasting/test_prophet.py

```diff
@@ -234,3 +241,39 @@ def helper_test_prophet_model(self, period, freq, compare_all_models=False):
         for pred in compare_preds:
             for val_i, pred_i in zip(val.univariate_values(), pred.univariate_values()):
                 self.assertAlmostEqual(val_i, pred_i, delta=0.1)
+
+    def test_conditional_seasonality(self):
+        """
+        Test that conditional seasonality is correctly incorporated by the model
+        """
+        df = pd.DataFrame()
+        df["ds"] = pd.date_range(start="2022-01-02", periods=395)
+        df["y"] = [i + 10 * (i % 7 == 0) for i in range(395)]
+        df["is_sunday"] = df["ds"].apply(lambda x: int(x.weekday() == 6))
+
+        ts = TimeSeries.from_dataframe(
+            df[:-30], time_col="ds", value_cols="y", freq="D"
+        )
+        future_covariates = TimeSeries.from_dataframe(
+            df, time_col="ds", value_cols=["is_sunday"], freq="D"
+        )
+        expected_result = TimeSeries.from_dataframe(
+            df[-30:], time_col="ds", value_cols="y", freq="D"
+        )
+
+        model = Prophet(seasonality_mode="additive")
+        model.add_seasonality(
+            name="weekly_sun",
+            seasonal_periods=7,
+            fourier_order=2,
+            condition_name="is_sunday",
+        )
+
+        model.fit(ts, future_covariates=future_covariates)
+
+        forecast = model.predict(30, future_covariates=future_covariates)
```

Comment:
also here, predicting 7 days should already be enough

Reply:
Done